client/environment_test.go

```diff
@@ -45,11 +45,88 @@ var _ = Describe("Environment Client", func() {
 			})
 		})
 
+		Describe("SuccessMultiPages", func() {
+			var environmentsP1, environmentsP2 []Environment
+			for i := 0; i < 100; i++ {
+				environmentsP1 = append(environmentsP1, mockEnvironment)
+			}
+
+			for i := 0; i < 33; i++ {
+				environmentsP2 = append(environmentsP2, mockEnvironment)
+			}
+
+			BeforeEach(func() {
+				httpCall = mockHttpClient.EXPECT().
+					Get("/environments", map[string]string{
+						"offset": "0",
+						"limit":  "100",
+					}, gomock.Any()).
+					Do(func(path string, request interface{}, response *[]Environment) {
+						*response = environmentsP1
+					}).Times(1)
+
+				httpCall2 = mockHttpClient.EXPECT().
+					Get("/environments", map[string]string{
+						"offset": "100",
+						"limit":  "100",
+					}, gomock.Any()).
+					Do(func(path string, request interface{}, response *[]Environment) {
+						*response = environmentsP2
+					}).Times(1)
+
+				environments, err = apiClient.Environments()
+			})
+
+			It("Should return the environments", func() {
+				Expect(environments).To(Equal(append(environmentsP1, environmentsP2...)))
+			})
+		})
+
+		Describe("SuccessMultiPagesWithProject", func() {
```

Comment:
should have one with organization as well, and one that should FAIL when both are passed

Reply:
This is not something that is enforced in the code. Therefore, not really something we would want to test.
The backend API will reject it.